tests/test_cli.py

```diff
@@ -239,7 +239,37 @@ def test_image_files_overwritten_if_no_output_dir(integration):
     assert shape == (500, 500, 3)
 
 
-def test_main_ImageReadError():
-    """Monketpatch an ImageReadError to main to trigger those lines"""
-    # TODO: that line of inquiry should also reject_count += 1
-    pass
+@pytest.mark.parametrize(
+    "extension, error_expected, expected",
+    [
+        (".png", False, "png"),
+        ("png", False, "png"),
+        ("PNG", False, "png"),
+        ("fake_ext", True, None),
+        (".fake_ext", True, None),
+        ("", True, None),
+    ],
+)
+def test_check_extension(extension, error_expected, expected):
+    if error_expected:
+        pytest.raises(argparse.ArgumentTypeError, chk_extension, extension)
+    else:
+        assert chk_extension(extension) == expected
+
+
+@pytest.mark.slow
+def test_extension_parameter(integration):
+    sys.argv = [
+        "",
+        "-i",
+        "tests/test",
+        "-o",
+        "tests/crop",
+        "-r",
+        "tests/reject",
+        "-e",
+        "png",
+    ]
+    command_line_interface()
```

Comment:
And here I want to make sure that all images have the intended extension if I'm using the `--extension` flag. So I set up a call as though it came from the shell: [you can read more about this here](https://www.geeksforgeeks.org/how-to-use-sys-argv-in-python/).

After that, all that needs to be done is to make sure that every file ends with the `png` extension.